src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useMemo } from 'react';
+import { ISearchInputProps } from './interfaces';
+import * as Icon from '../Icons';
+import * as S from './styles';
+import { debounce } from './utils';
+import { farmerConnectTheme } from '../Theme';
+
+const SearchInput = ({
+  className,
+  value,
+  onChange,
+  onClear,
+  onSearch,
+  debounceTime = 0,
+  placeholder = 'Search here...',
+  ...props
```

Comment:
API looks good!

If you want to add in the suggestions in this iteration I would suggest making it an array of objects that is passed into the input and then is simply rendered. Totally fine to leave it out though as we don't need it in our first usage.

Reply:
I believe we will leave the suggestions for later as we don't need it for now. Thanks for the tip though!